src/random.cpp

```diff
@@ -65,6 +65,34 @@ static inline int64_t GetPerformanceCounter()
 #endif
 }
 
+#if !defined(_MSC_VER) && (defined(__x86_64__) || defined(__amd64__))
+static bool RDRandSupported()
+{
+    uint32_t ax,bx,cx,dx,func=1;
+    __asm__ volatile ("cpuid": "=a" (ax), "=b" (bx), "=c" (cx), "=d" (dx) : "a" (func));
+    return (cx & 0x40000000);
+}
+static const bool fRDRand = RDRandSupported();
```

Comment:
I believe that C++11 specifies something about the order of initialization of statics defined inside functions, which results in automatically emitted synchronization calls. Perhaps that's not the case for const fields, though.